### PR TITLE
[Storage] convert options to take pointer

### PR DESCRIPTION
**Pull request by seankane-msft**
Converts `DownloadResponse.Body` to take a pointer to an options struct.

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

File: sdk/storage/azblob/zc_response_helpers.go [hunk 1]
```diff
@@ -50,11 +50,14 @@ type DownloadResponse struct {
 // while reading, it will make additional requests to reestablish a connection and
 // continue reading. Specifying a RetryReaderOption's with MaxRetryRequests set to 0
 // (the default), returns the original response body and no retries will be performed.
```
Comment (kimprice) [inline sdk/storage/azblob/zc_response_helpers.go:52]: Should this comment include something about passing in 'nil' to accept the default options?

Comment (seankane-msft) [reply sdk/storage/azblob/zc_response_helpers.go:52]: Yes it should, added!

**Review by jhendrixMSFT — APPROVED** [review]

**Review by kimprice — APPROVED** [review]

**Review by mohsha-msft — APPROVED** [review]
:shipit:

**Comment by mohsha-msft** [comment]
I've created a release candidate for azblob 0.3.0 [here](https://github.com/Azure/azure-sdk-for-go/pull/16989). You can merge your branch to this one.

**Pull request merged**